smarts/core/smarts.py

```diff
@@ -181,6 +181,13 @@ def step(self, agent_actions):
             self.destroy()
             raise  # re-raise
 
+    def _check_if_acting_on_active_agents(self, agent_actions):
+        for agent_id in agent_actions.keys():
+            if agent_id not in self._agent_manager.ego_agent_ids:
+                self._log.warning(
+                    f"Attempted giving action to an non-existing agent, {agent_id} "
```

Comment:
Maybe use `perform` instead of `give`? But I am not sure whether to use to do or doing here :smile: